astropy/io/fits/tests/test_convenience.py

```diff
@@ -255,6 +271,51 @@ def test_tabledump(self):
         fits.tabledump(self.temp('tb.fits'), datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
 
+        # test with datafile and cdfile
+        datafile = self.temp('data.txt')
+        cdfile = self.temp('coldefs.txt')
+        fits.tabledump(self.temp('tb.fits'), datafile, cdfile)
+        assert os.path.isfile(datafile)
+        with open(datafile) as data:
+            assert data.read() == datastr
+        with open(cdfile) as coldefs:
+            assert coldefs.read() == cdstr
+
+    @pytest.mark.parametrize('tablename', ['table.fits', 'tb.fits'])
+    def test_dump_load_round_trip(self, tablename):
+        """
+        A simple test of the dump/load methods; dump the data, column, and
+        header files and try to reload the table from them.
+        """
+
+        def _assert_attr_col(tbhdu, fitsfile):
+            """
+            Helper function to compare column attributes
+            """
+            with fits.open(fitsfile) as hdul:
+                # Double check that the headers are equivalent
+                assert hdul[1].columns.names == tbhdu.columns.names
+                attrs = [k for k, v in fits.Column.__dict__.items()
+                         if isinstance(v, ColumnAttribute)]
+                for name in hdul[1].columns.names:
+                    col = hdul[1].columns[name]
+                    new_col = tbhdu.columns[name]
+                    for attr in attrs:
+                        if getattr(col, attr) and getattr(new_col, attr):
+                            assert getattr(col, attr) == getattr(new_col, attr)
```

Comment:
To avoid the duplication, could you put the function in `test_table` and import it here ?